shared/protocol/src/main/java/io/pravega/shared/NameUtils.java

```diff
@@ -227,6 +228,7 @@ public static String getTransientNameFromId(String parentStreamSegmentName, UUID
      *
      * @return The start position of the delimiter contained within streamSegmentName.
      */
+    @NonNull
```

Comment:
I'm not sure what this annotation would do here. You're annotating the method, but the method can't possibly return null.